pkgs/applications/editors/neovim/default.nix

```diff
@@ -119,7 +121,18 @@ in
       )
     '' + lib.optionalString stdenv.isDarwin ''
       substituteInPlace src/nvim/CMakeLists.txt --replace "    util" ""
-    '';
+    '' + ''
+      mkdir -p $out/lib/nvim/parser
+    '' + lib.concatStrings (lib.mapAttrsToList
+      (language: src: ''
+        ln -s \
+          ${tree-sitter.buildGrammar {
+            inherit language src;
+            version = "neovim-${version}";
+          }}/parser \
+          $out/lib/nvim/parser/${language}.so
```

Comment:
shouldn't you link the queries as well ?

Reply:
idk, neovim comes with its own queries: https://github.com/neovim/neovim/tree/v0.9.0/runtime/queries

Reply:
is $out/lib/nvim  in runtimepath directory ? seems like it can't find the grammars, `:help lua` doesn't give me syntax highlighting. `$out/share/nvim/runtime/parser`  should work. I can't help but feel like you are going to prove me wrong once again xD

Reply:
IIUC that's where the cmake file installs to: https://github.com/neovim/neovim/blob/f17bb4f41102ecec7989bc4c14c626dc595e2f0b/cmake.deps/cmake/TreesitterParserCMakeLists.txt#L19

Also dundar's PR checked for `nvim/parser`, and `$out/share/nvim/runtime/parser` might not have worked: https://github.com/neovim/neovim/pull/22923/files#diff-1f1596d636df8da9babb14a7e4f135458df238e66dcccb4bee2a002eda13ac68R39